libs/core/src/cart/testing/factories/cart.factory.ts

```diff
@@ -8,9 +8,23 @@ import { CartShippingRate } from '../../model/cart-shipping-rate';
 
 @Injectable()
 export class CartFactory {
+
+  cart:MockCart = {...new MockCart()};
   
   create() : Cart {
-    return {...new MockCart()};
+    return this.cart;
+  }
+
+  createCartItem() : CartItem {
+    return new MockCartItem();
+  }
+
+  addCartItemToCart(reqBody) {
```

Comment:
This doesn't make sense as a factory method. What's the purpose of this method?

Reply:
The purpose was to simulate a backend that would be responsible for adding a cartItem to the cart, then returning the new cart. Should this be in the testing service instead?

Reply:
@lderrickable I'd say yes. The factory's responsibility is for generating new instances of an object easily. Currently, you're utilizing it to maintain state. If we could make the class readonly, that would enforce the condition I'm describing.

Reply:
done